src/libraries/System.Text.Json/src/System/Text/Json/JsonHelpers.Escaping.cs

```diff
@@ -53,6 +53,34 @@ public static byte[] GetEscapedPropertyNameSection(ReadOnlySpan<byte> utf8Value,
             return escapedString;
         }
 
+        public static string GetPropertyName(
+            string rawPropertyName,
+            JavaScriptEncoder? encoder)
+        {
+            int indexOfFirstCharacterToEncode;
+            string returnValue;
+
+            indexOfFirstCharacterToEncode = JsonWriterHelper.NeedsEscaping(rawPropertyName.AsSpan(), encoder);
```

Comment:
Given that this method is only invoked after a check to `propertyName.AsSpan().ContainsSpecialCharacters()` this check seems a bit redundant to me. It seems like this could be replaced with
```suggestion
            Debug.Assert(JsonWriterHelper.NeedsEscaping(rawPropertyName.AsSpan(), encoder) != -1);
```
And the method renamed to `GetEscapedPropertyName` (like the helper you have defined below).

Reply:
> Given that this method is only invoked after a check to `propertyName.AsSpan().ContainsSpecialCharacters()` this check seems a bit redundant to me. It seems like this could be replaced with
> 
> And the method renamed to `GetEscapedPropertyName` (like the helper you have defined below).

I fixed it by https://github.com/dotnet/runtime/pull/84688/commits/a610e38ee705bfe38420a1993af2ffc6b8101186 , but after that **GetPath_Special Characters** test was failed with next error message:

> [xUnit.net 00:06:26.33]     System.Text.Json.Nodes.Tests.ParentPathRootTests.GetPath_SpecialCharacters [FAIL]
>   Failed System.Text.Json.Nodes.Tests.ParentPathRootTests.GetPath_SpecialCharacters [626 ms]
>   Error Message:
>    Microsoft.VisualStudio.TestPlatform.TestHost.DebugAssertException : Method Debug.Fail failed with '
> ', and was translated to Microsoft.VisualStudio.TestPlatform.TestHost.DebugAssertException to avoid terminating the process hosting the test.
>   Stack Trace:
>      at Microsoft.VisualStudio.TestPlatform.TestHost.TestHostTraceListener.GetException(String message)
>    at Microsoft.VisualStudio.TestPlatform.TestHost.TestHostTraceListener.Fail(String message, String detailMessage)
>    at System.Diagnostics.TraceInternal.Fail(String message, String detailMessage) in C:\runtime\src\libraries\System.Diagnostics.TraceSource\src\System\Diagnostics\TraceInternal.cs:line 262
>    at System.Diagnostics.TraceInternal.TraceProvider.Fail(String message, String detailMessage) in C:\runtime\src\libraries\System.Diagnostics.TraceSource\src\System\Diagnostics\TraceInternal.cs:line 17
>    at System.Diagnostics.Debug.Fail(String message, String detailMessage) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Diagnostics\Debug.cs:line 133
>    at System.Diagnostics.Debug.Assert(Boolean condition, String message, String detailMessage) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Diagnostics\Debug.cs:line 97
>    at System.Diagnostics.Debug.Assert(Boolean condition) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Diagnostics\Debug.cs:line 82
>    at System.Text.Json.JsonHelpers.GetEscapedPropertyName(String rawPropertyName, JavaScriptEncoder encoder) in C:\runtime\src\libraries\System.Text.Json\src\System\Text\Json\JsonHelpers.Escaping.cs:line 60
>    at System.Text.Json.Nodes.JsonObject.GetPath(ValueStringBuilder& path, JsonNode child) in C:\runtime\src\libraries\System.Text.Json\src\System\Text\Json\Nodes\JsonObject.cs:line 210
>    at System.Text.Json.Nodes.JsonValue.GetPath(ValueStringBuilder& path, JsonNode child) in C:\runtime\src\libraries\System.Text.Json\src\System\Text\Json\Nodes\JsonValue.cs:line 97
>    at System.Text.Json.Nodes.JsonNode.GetPath() in C:\runtime\src\libraries\System.Text.Json\src\System\Text\Json\Nodes\JsonNode.cs:line 136
>    at System.Text.Json.Nodes.Tests.ParentPathRootTests.GetPath_SpecialCharacters() in C:\runtime\src\libraries\System.Text.Json\tests\System.Text.Json.Tests\JsonNode\ParentPathRootTests.cs:line 99
>    at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
>    at System.Reflection.MethodBaseInvoker.InterpretedInvoke_Method(Object obj, IntPtr* args) in C:\runtime\src\coreclr\System.Private.CoreLib\src\System\Reflection\MethodBaseInvoker.CoreCLR.cs:line 36
>    at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Reflection\MethodBaseInvoker.cs:line 57
>    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Reflection\RuntimeMethodInfo.cs:line 127
>    at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Reflection\MethodBase.cs:line 56
>    at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\AsyncMethodBuilderCore.cs:line 38
>    at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\AsyncMethodBuilderCore.cs:line 38
>    at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\AsyncMethodBuilderCore.cs:line 38
>    at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\AsyncMethodBuilderCore.cs:line 38
>    at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\AsyncMethodBuilderCore.cs:line 38
>    at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\AsyncMethodBuilderCore.cs:line 38
>    at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\AsyncMethodBuilderCore.cs:line 38
>    at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\AsyncMethodBuilderCore.cs:line 38
>    at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\AsyncMethodBuilderCore.cs:line 38
>    at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\AsyncMethodBuilderCore.cs:line 38                                                                                                                                                                                                  at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\AsyncMethodBuilderCore.cs:line 38                                                                                                                                                                                                  at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\AsyncMethodBuilderCore.cs:line 38                                                                                                                                                                                          10:19   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\AsyncMethodBuilderCore.cs:line 38                                                                                                                                                                                                  at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\AsyncMethodBuilderCore.cs:line 38                                                                                                                                                                                                  at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\AsyncMethodBuilderCore.cs:line 38                                                                                                                                                                                                  at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Runtime\CompilerServices\AsyncMethodBuilderCore.cs:line 38                                                                                                                                                                                                  at System.Threading.Tasks.Task`1.InnerInvoke() in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Threading\Tasks\Future.cs:line 501                                                                    at System.Threading.Tasks.Task.<>c.<.cctor>b__281_0(Object obj) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Threading\Tasks\Task.cs:line 2387                                                    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Threading\ExecutionContext.cs:line 179                                                                                                                                                                                                      at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Threading\Tasks\Task.cs:line 2345              at System.Threading.Tasks.Task.ExecuteEntry() in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Threading\Tasks\Task.cs:line 2258                                                                      at System.Threading.Tasks.SynchronizationContextTaskScheduler.<>c.<.cctor>b__8_0(Object s) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Threading\Tasks\TaskScheduler.cs:line 606                 at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Threading\ExecutionContext.cs:line 179                                                                                                                                                                                                      at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Threading\ExecutionContext.cs:line 143                                                                                                                                                                                                              at System.Threading.Tasks.Task.InnerInvoke() in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Threading\Tasks\Task.cs:line 2405                                                                       at System.Threading.Tasks.Task.<>c.<.cctor>b__281_0(Object obj) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Threading\Tasks\Task.cs:line 2387                                                    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Threading\ExecutionContext.cs:line 179                                                                                                                                                                                                      at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Threading\Tasks\Task.cs:line 2345              at System.Threading.Tasks.Task.ExecuteEntryUnsafe(Thread threadPoolThread) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Threading\Tasks\Task.cs:line 2283                                         at System.Threading.Tasks.ThreadPoolTaskScheduler.<>c.<.cctor>b__10_0(Object s) in C:\runtime\src\libraries\System.Private.CoreLib\src\System\Threading\Tasks\ThreadPoolTaskScheduler.cs:line 35                   at System.Threading.Thread.StartCallback() in C:\runtime\src\coreclr\System.Private.CoreLib\src\System\Threading\Thread.CoreCLR.cs:line 104

That's why I made revert https://github.com/dotnet/runtime/pull/84688/commits/4a0d7b17c4ad86dbe77fce5281a41bbb175e53d9